File: backend/schedule-service/src/main/java/com/schedulerates/schedule/model/schedule/Schedule.java
package com.schedulerates.schedule.model.schedule;

import com.schedulerates.schedule.model.common.BaseDomainModel;
import lombok.*;
import lombok.experimental.SuperBuilder;

/**
 * Represents a domain model for a Transportation as {@link Transportation}.
 */
@Getter
@Setter
@SuperBuilder
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
public class Schedule extends BaseDomainModel {


    private String id;

    // Foreign keys
    private String portFromId;
    private String portToId;
    private String companyId;

    private String portFromName;
    private String portToName;
    private String companyName;

    // Schedule information
    private String dateDepart;
    private String dateArrive;
    private Integer transit;

    // Vessel and service info
    private String vessel;
    private String refVoyage;
    private String serviceName;

}

File: frontend/src/app/users/modules/schedule.ts
export interface Schedule {
  id: string;
  portFromId: string;
  portToId: string;
  companyId: string;
  portFromName: string;
  portToName: string;
  companyName: string;
  dateDepart: string;
  dateArrive: string;
  transit: number;
  vessel: string;
  refVoyage: string;
  serviceName: string;
  active: number;
}
